testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/DeleteAccountTest.java

```diff
@@ -0,0 +1,210 @@
+    //reset role back since realm is shared among tests
+    disableDeleteAccountRequiredAction();
+  }
+
+
+  @Test
+  public void deleteOwnAccountPageNotVisibleAndNotAccessibleWithoutDeleteAccountActionEnabled() {
+    addDeleteAccountRoleToUserClientRoles();
+    List<WebElement>  accountLinks = driver.findElements(By.cssSelector("div[id^=\"landing-\"]"));
+    Assert.assertTrue(accountLinks.stream().noneMatch(link -> link.getAttribute("id").contains("delete-account")));
+    deleteAccountPage.navigateTo();
+    pageNotFound.assertCurrent();
+    removeDeleteAccountRoleToUserClientRoles();
+  }
+
+  @Test
+  public void deleteOwnAccountAIACancellationSucceeds() {
+    enableDeleteAccountRequiredAction();
+    addDeleteAccountRoleToUserClientRoles();
+    deleteAccountPage.navigateTo();
+    deleteAccountPage.getDeleteAccountButton().click();
+    loginToAccount();
+    URI uri = URI.create(driver.getCurrentUrl());
+    Assert.assertTrue(uri.getPath().contains("login-actions"));
+    Assert.assertTrue(uri.getQuery().contains("execution=delete_account"));
+    driver.findElement(By.cssSelector("button[name='cancel-aia']")).click();
+    Assert.assertTrue(deleteAccountPage.isCurrent());
+    disableDeleteAccountRequiredAction();
+    removeDeleteAccountRoleToUserClientRoles();
+  }
+
+  @Test
+  public void deleteOwnAccountForbiddenWithoutDeleteAccountActionEnabled() {
+    enableDeleteAccountRequiredAction();
+    addDeleteAccountRoleToUserClientRoles();
+    deleteAccountPage.navigateTo();
+    deleteAccountPage.getDeleteAccountButton().click();
+    loginToAccount();
+    disableDeleteAccountRequiredAction();
+    driver.findElement(By.cssSelector("input[type='submit']")).click();
+    Assert.assertEquals(driver.findElements(By.cssSelector(".alert-error")).size(), 1);
+    Assert.assertEquals(driver.findElement(By.cssSelector("#kc-content-wrapper > div > span.kc-feedback-text")).getText(), "You do not have enough permissions to delete your own account, contact admin.");
+    removeDeleteAccountRoleToUserClientRoles();
+  }
+
+  @Test
+  public void deleteOwnAccountSucceeds() {
+    String userId = createUserToBeDeleted("test-user-to-be-deleted@localhost", "password");
+    enableDeleteAccountRequiredAction();
+    addDeleteAccountRoleToUserClientRoles(userId);
+    addDeleteAccountRoleToUserClientRoles();
+    deleteAccountPage.navigateTo();
+    deleteAccountPage.header().clickLogoutBtn();
+    welcomeScreen.header().clickLoginBtn();
+    loginPage.form().login("test-user-to-be-deleted@localhost", "password");
+    deleteAccountPage.navigateTo();
+    Assert.assertTrue(deleteAccountPage.isCurrent());
+    deleteAccountPage.getDeleteAccountButton().click();
+    driver.findElement(By.id("username")).sendKeys("test-user-to-be-deleted@localhost");
+    driver.findElement(By.id("password")).sendKeys("password");
+    driver.findElement(By.id("kc-login")).click();
+    driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
```

Comment:
Why is this necessary?

Reply:
For some reason, the test was not stable, sometimes it fails sometimes it succeeds. Now, the need for this has gone once I created a new page with selectors.